src/lib.rs

```diff
@@ -163,6 +165,29 @@ impl Order {
         self.account.get(&mut self.nonce, challenge_url).await
     }
 
+    /// Poll the order with exponential backoff until in a final state
+    ///
+    /// Refresh the order state from the server for `tries` times, waiting `delay` before the
+    /// first attempt and increasing the delay by a factor of 2 for each subsequent attempt.
+    ///
+    /// Yields the [`OrderStatus`] immediately if final or after `tries` attempts.
+    ///
+    /// (Empirically, we've had good results with 5 tries and an initial delay of 250ms.)
+    pub async fn poll(&mut self, mut tries: u8, mut delay: Duration) -> Result<OrderStatus, Error> {
+        loop {
+            sleep(delay).await;
+            let state = self.refresh().await?;
```

Comment:
Not sure whether we should keep more trace events in here? Turns out we don't currently have a library-level dependency on tracing here at all, so it's probably orthogonal?

Reply:
I feel like it would be useful operationally if it were exposed at a `Trace` level. It feels like the kind of thing you don't care about at all until you're down a debugging rabbit hole and then it might be important (e.g. when trying to figure out rate limit issues).

Reply:
Fair -- let's postpone until we decide to add more tracing more broadly.